### PR TITLE
fix(#7456): allow children of Overlay Plots to be removed

### LINKED ISSUE #7456: [Overlay Plots] Cannot Remove Elements from a Plot
#### Summary

We currently cannot remove elements from an Overlay Plot if the plot is in view. If we attempt to remove it from the Element Pool or through the Object Tree, this error is thrown: 
Screenshot 2024-02-02 at 9 54 52 AM

The only workaround is to navigate _away_ from the object, and remove the elements through the Object Tree. 
#### Expected vs Current Behavior

We should be able to remove elements from a plot. 
#### Steps to Reproduce

1. Create or Navigate to an Overlay Plot with multiple elements in it. Open your console. 
2. Go to Edit Mode, and in the Elements Pool, right click one of the elements and press "remove". 
3. Observe that there is an error thrown in the console, similar to the image above. 
4. Save the object and refresh. Observe that the element you tried to remove, has indeed, NOT been removed. 
5. Now, go to the object tree and try to remove an element from the Overlay Plot from there. 
6. Repeat 3 and 4. 

#### Environment

* Open MCT Version:  3.3.0
* Deployment Type:  /testathon, commit hash 5e3b051bf150a98eebf2cf4ed39bd9caf4b8c017
* OS:
* Browser: Chrome

#### Impact Check List

- [x] Data loss or misrepresented data?
- [x] Regression? Did this used to work or has it always been broken?
- [x] Is there a workaround available?
- [x] Does this impact a critical component?
- [ ] Is this just a visual bug with no functional impact?
- [ ] Does this block the execution of e2e tests?
- [ ] Does this have an impact on Performance?

### PR DESCRIPTION
**Pull request by ozyx**
Closes #7456 

### Describe your changes:

The StalenessMixin assumes that it will always be passed a `domainObject` when looking to unsubscribe from staleness. However, in the case of Overlay Plots, it receives an Identifier directly, causing an error. This modifies that logic to check if its an Identifier first before attempting to extract an Identifier from the object.

- Refactors `objectUtils` to have ESM exports of individual helper functions
- Refactors usage of `objectUtils` helper functions to use ESM imports
- Refactors some Vue anti patterns out of PlotElementsPool.vue

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [ ] Tests included and/or updated with changes?
* [x] Has this been smoke tested?
* [x] Have you associated this PR with a `type:` label? Note: this is not necessarily the same as the original issue.
* [x] Have you associated a milestone with this PR? Note: leave blank if unsure.
* [ ] Is this a breaking change to be called out in the release notes?
* [ ] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [ ] Changes appear to address issue?
* [ ] Reviewer has tested changes by following the provided instructions?
* [ ] Changes appear not to be breaking changes?
* [ ] Appropriate automated tests included?
* [ ] Code style and in-line documentation are appropriate?

### CODE REVIEW

File: e2e/tests/functional/plugins/plot/overlayPlot.e2e.spec.js [hunk 1]
```diff
@@ -315,6 +315,31 @@ test.describe('Overlay Plot', () => {
...
+
+    const swgA = await createDomainObjectWithDefaults(page, {
+      type: 'Sine Wave Generator',
+      parent: overlayPlot.uuid
+    });
+
+    await page.goto(overlayPlot.url);
+    // Wait for plot series data to load and be drawn
+    await waitForPlotsToRender(page);
+    await page.getByTitle('Edit Object').click();
```
Comment (unlikelyzero) [inline e2e/tests/functional/plugins/plot/overlayPlot.e2e.spec.js:332]: can we work around the Title?

**Review by unlikelyzero — APPROVED** [review]
Two comments

File: e2e/tests/functional/plugins/plot/overlayPlot.e2e.spec.js (hunk 1 above)
Comment (ozyx) [reply e2e/tests/functional/plugins/plot/overlayPlot.e2e.spec.js:332]: whoop, yeah let me see if I can update this

File: e2e/tests/functional/plugins/plot/overlayPlot.e2e.spec.js [hunk 2]
```diff
@@ -315,6 +315,31 @@ test.describe('Overlay Plot', () => {
       expect(plotPixelSize).toBeGreaterThan(0);
     }
   );
+
+  test('Can remove an item via the elements pool action menu', async ({ page }) => {
+    const overlayPlot = await createDomainObjectWithDefaults(page, {
```
[the remark this answers is not in the document]
Comment (ozyx) [reply e2e/tests/functional/plugins/plot/overlayPlot.e2e.spec.js]: done!

**Pull request merged**